packages/trader/src/Modules/Trading/Components/Elements/purchase-button.tsx

```diff
@@ -1,13 +1,42 @@
 import classNames from 'classnames';
-import PropTypes from 'prop-types';
 import React from 'react';
 import { DesktopWrapper, MobileWrapper, Money, IconTradeTypes, Text } from '@deriv/components';
 import { getContractTypeDisplay } from 'Constants/contract';
 import ContractInfo from 'Modules/Trading/Components/Form/Purchase/contract-info.jsx';
 import { getGrowthRatePercentage } from '@deriv/shared';
+import { TProposalTypeInfo } from '../../../../../types';
+
+type TPurchaseButton = {
```

Comment:
can you take a look at the locations where <PurchaseButton /> is used and make some of the props optional?

Otherwise there'll be error in files where the component is used.

Reply:
`PurchaseButton` is used only in `PurchaseFieldset`👍🏻